File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n=nums.length, a=0, b=0,count=0;
        while(b<n){
            if(nums[b]==0){
                b++;
                count++;
            }
            else if(nums[b]!=0){
                int temp=nums[a];
                nums[a]=nums[b];
                nums[b]=temp;
                a++;
                b++;
            }
        }
        for(int i=n-count;i<n;i++){
            nums[i]=0;
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        L=0
        R=1
        n=len(nums)

        while (R<n and L<n):
            if(nums[L]!=0):
                L+=1
                R=L
            elif(nums[R]==0):
                R+=1
            else:
                temp=nums[L]
                nums[L]=nums[R]
                nums[R]=temp
        